SQLServer.md

```diff
@@ -0,0 +1,13 @@
+##SQL Server dialect  
+
+### Limitations
+
+* Batch insert actually inserts rows one by one, because SQLServer do not return generated keys.  
```

Comment:
Is it possible to workaround that behavior somehow? 
For example to replace jdbc batching with single insert sql statement with OUTPUT?
I afraid that instead users will get huge performance degradation when they will use Exposed DAO .
```SQL 
INSERT INTO Cities (name) OUTPUT inserted.id  VALUES ('London'), ('Moscow');
```

Reply:
Yes, this is performance degradation. Actually, batch insert works fine, it doesn't return keys and current API do require generated key. So we can do API degradation, instead of performance degradation.

Reply:
What do you think if we'll use "specific" BatchInsertStatement class for SQLServer like [this](https://gist.github.com/Tapac/14201dbc20a7a17c5df92ac543d2b4dc). I ran tests and they all passed.

Reply:
Looks good, but I think that we need to measure performance gain.

Reply:
Could you please make some benchmarks for both approaches for different batch and column sizes?
I don't think that it's required to be commit to Exposed, but if you want you may do it under tests in a separate package (similar to MysqlTests.kt)

Reply:
Oops, I am afraid my 8 Gb Mac is no good for performance tests, I am afraid it can take some time from my side. I suggest to start with current implementation and improve afterwards.

Reply:
Finally, I tested batch performance vs. single inserts and yes, I see performance degradation from 12 seconds to 1 minute (Windows PC/SSD/fresh SQL Server 2017). Unfortunately, I have no idea how to combine batche performance and return generated keys.

Maybe we can add special method that makes insert without returning generated keys?

Reply:
I do some perf tests too and looks like approach without OUTPUT works better then one-by-one insertions.
Could you look at https://gist.github.com/anonymous/63ddbf72328d9fd032d98475f7c0b45d and verify that I don't make any mistakes attempting to create perf-test. 

I think that randomization with nullable columns not very fair...

Reply:
@Tapac Yes, I can confirm that OUPUT approaches works better than single inserts, but worse than normal batch. My numbers are 1 minute for inserts, 4 seconds for output and 0.2 seconds for batch. 

Unfortunately, OUTPUT approach accepts not more than 1000 rows and not more than 2100 SQL parameters.

Reply:
Interesting limitation, I will handle it when attempting to add new batch and flush previous value if current state exceed limit. Something like batch of batches.

Reply:
Of cause we can do it, but this is dirty hack. And I am sure that for large batches nobody cares about generated keys. But from performance point of view batch approach is 10 times faster. And support for inserts without generated keys can be used for all other database.

Reply:
Batch insert is used on flushing Exposed entities to DB and then retrieve generated id. Creating a lot of entities is a common case for any application, so we can't make it one-by-one.